spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/route/TimeoutConfig.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.gateway.route;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.springframework.validation.annotation.Validated;
+
+/**
+ * Configuration for per-route http timeouts.
+ *
+ * @author Stefan Stus
+ */
+@Validated
+public class TimeoutConfig {
+
+	public static final TimeoutConfig EMPTY_TIMEOUT_CONFIG = new TimeoutConfig();
+
+	/** The connect timeout in millis */
+	private Optional<Integer> connectTimeout;
```

Comment:
`Optional` is considered a bad practice to be used as a field

Reply:
removed and made fields nullable